### DISCUSSION TITLE
Questions regarding the refreshable annotation

### DISCUSSION THREAD

**Discussion by LarsEggimann** [zauberzeug/nicegui#2806] [Q&A]
### Question

Hello there!

I am using niceGUI for a project at uni and am using the @ui.refreshable annotation to re-render certain elements. In my app i use a setup similar to this:
```python
from nicegui import ui

class Test:
    @ui.refreshable
    def some_test(self):
        var = "Text from some_test() function"
        ui.label(var)

test = Test()
test.some_test()

def refresh_some_test():
    test.some_test.refresh()
    
ui.button("Refresh", on_click=refresh_some_test)

ui.run()
```

now when i run this there are no errors, but my code editor (pylint on vs-code) gives the following error on the line `test.some_test.refresh()`:
`Method 'some_test' has no 'refresh' memberPylint[E1101:no-member](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/no-member.html)`

My question is: am using the niceGUI library wrong or is this an issue with pylint not recognizing the members correctly?

Maybe there is a much easier way to achieve this then I would also like to hear some suggestions. 

thanks in advance for all the help.

Cheers,
Lars

**Comment by falkoschindler** [comment]
Hi @LarsEggimann,

This problem has been discussed in issue https://github.com/zauberzeug/nicegui/issues/2174. The error you're referring to seemed to come from a pylint bug (see https://github.com/pylint-dev/pylint/issues/9246). But even after upgrading to version >=3.0.4 the problem persists. So I actually don't know how to avoid it other than adding a `# noqa` or `# type: ignore` marker. 😕

**Reply by LarsEggimann** [reply]
Hi @falkoschindler 
Thank you for the very fast answer.
I'm sorry for asking duplicate questions, I did a quick search but is seems that I missed the issue you mentioned.

Then i will just add `# pylint: disable=no-member` as a temporary solution. 

Thanks again!
